store/migration/dev/20230117000000##idp.sql

```diff
@@ -0,0 +1,22 @@
+-- idp stores generic identity provider.
+CREATE TABLE idp (
+  id SERIAL PRIMARY KEY,
+  row_status row_status NOT NULL DEFAULT 'NORMAL',
+  creator_id INTEGER NOT NULL REFERENCES principal (id),
+  created_ts BIGINT NOT NULL DEFAULT extract(epoch from now()),
+  updater_id INTEGER NOT NULL REFERENCES principal (id),
+  updated_ts BIGINT NOT NULL DEFAULT extract(epoch from now()),
+  resource_id TEXT NOT NULL,
+  name TEXT NOT NULL,
+  type TEXT NOT NULL CONSTRAINT idp_type_check CHECK (type IN ('OAUTH2', 'OIDC')),
+  -- config stores the corresponding configuration of the IdP, which may vary depending on the type of the IdP.
+  config JSONB NOT NULL
+);
+
```

Comment:
add ALTER SEQUENCE idp_id_seq RESTART WITH 101;

Reserve 1 ~ 100 in case we need to have some internal idp

Reply:
Updated